### PR TITLE
Timeout of 600s added to ExternalProjects

### PR DESCRIPTION
**Pull request by Harshalzzzzzzz**
@oshadura @Axel-Naumann 
Dear Mentors,
Here's my pull request for the issue https://github.com/root-project/root/issues/6635.
Can you please review this. Please excuse any instances where the timeout parameter has been added unnecessarily as it wasn't clear to me how to test this. Since the desired interval for `TIMEOUT` hadn't been mentioned, I have added it as 600 s.
I'll fix any incorrect changes as soon as review comments and suggestions are added. I will squash the commits after the review.

Thanks in advance !

### CODE REVIEW

File: cmake/modules/SearchInstalledSoftware.cmake [hunk 1]
```diff
@@ -215,7 +220,9 @@ if(builtin_lzma)
       CONFIGURE_COMMAND ""
       BUILD_COMMAND ""
       INSTALL_COMMAND ""
-      LOG_DOWNLOAD 1)
+      LOG_DOWNLOAD 1
+      TIMEOUT ${timeoutProject}
```
Comment (Axel-Naumann) [inline cmake/modules/SearchInstalledSoftware.cmake:224]: We only need the `TIMEOUT` parameter for externals that we download; this one is picked up from within the source tree. IIUC `TIMEOUT` has no effect here. Please remove for all local URLs

File: builtins/davix/CMakeLists.txt [hunk 2]
```diff
@@ -51,7 +52,8 @@ ExternalProject_Add(DAVIX
     -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
     -DLIB_SUFFIX=
   LOG_BUILD 1 LOG_CONFIGURE 1 LOG_DOWNLOAD 1 LOG_INSTALL 1
-  BUILD_BYPRODUCTS ${DAVIX_LIBRARIES})
+  BUILD_BYPRODUCTS ${DAVIX_LIBRARIES}
+  TIMEOUT ${timeoutProject})
```
Comment (Axel-Naumann) [inline builtins/davix/CMakeLists.txt:56]: Why add a new variable, why not simply `TIMEOUT 600`? 600 sounds good btw, that's 10 minutes for download operations, good enough.

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]
Thanks! We'll need two changes:
- instead of using a variable for `600` everywhere, just use `600` everywhere.
- please remove the `TIMEOUT` parameter for all local files being built through `ExternalProject_Add`.

File: cmake/modules/SearchInstalledSoftware.cmake (hunk 1 above)
Comment (Harshalzzzzzzz) [reply cmake/modules/SearchInstalledSoftware.cmake:224]: @Axel-Naumann Thank you for the suggestions. 
Imm make the required changes.
I'll need to remove the parameters only in this file right?
https://github.com/root-project/root/blob/db09e093521829f1cb6aac7af194f681b1f19548/cmake/modules/SearchInstalledSoftware.cmake

**Pull request closed without merging**